python/singa/autograd.py

```diff
@@ -607,12 +607,17 @@ def set_params(self, **parameters):
         # examples: Layer.set_params(W=np.ones((in, out), dtype=np.float32)),
         # Layer.set_params(**{'block1':{'linear1':{'W':np.ones((in, out),
         # dtype=np.float32)}}})
-        for parameter_name in parameters:
-            assert isinstance(self.__dict__[parameter_name], Layer)
-            self.__dict__[parameter_name].set_params(
-                **parameters[parameter_name])
+        for (parameter_name, parameter_value) in parameters.items():
+            #assert isinstance(self.__dict__[parameter_name], Layer)
+            if isinstance(self.__dict__[parameter_name], Layer):
+                self.__dict__[parameter_name].set_params(
```

Comment:
self.__dict__[parameter_name] will not be a layer.
it is either a tensor/numpy or a dictionary.

Reply:
class block(autograd.Layer):
	    def __init__(self):
	        self.conv1=autograd.Conv2d(3,4,2)
	        self.conv2=autograd.Conv2d(4,4,2)

if we want to reset W of self.conv1. then we should send {'conv1':{'W': tensor}}
then parameter_name='conv1', parameter_value={'W':tensor}.
Since block.conv1 is a Layer. Then we call block.conv1.set_params(**{'W':tensor})

then parameter_name='W', parameter_value=tensor
Since block.conv1.W is a Tensor, Then we call set_one_params('W', tensor), to reset W